src/sass/partials/components/_forms.scss

```diff
@@ -1,26 +1,113 @@
-
-// Content Forms
 .base--FORM {
-  padding: 1em 0;
-
-  .form--field {
-    display: grid;
-    grid-template-columns: auto;
-    grid-template-rows: auto;
-    padding: .5em 0;
-  }
+  margin-top: 0;
 
   .form--alert {
-    color: color('red');
+    color: color('red', 50);
     font-size: .8em;
   }
 
   .form--description {
+    color: color('grey', 50);
     font-size: .8em;
-    padding: 5px 0;
+    font-style: italic;
+    margin-top: setting-get('rhythm') / 2;
+  }
+
+  .form--field {
+    margin-top: setting-get('rhythm') / .5;
+  }
+
+  .form--submit {
+    margin-top: setting-get('rhythm') / .5;
+  }
+
+  [type='date'],
+  [type='datetime'],
+  [type='datetime-local'],
+  [type='email'],
+  [type='month'],
+  [type='number'],
+  [type='password'],
+  [type='search'],
+  [type='tel'],
+  [type='text'],
+  [type='time'],
+  [type='url'],
+  [type='week'] {
+    @include input;
+  }
+
+  [type='checkbox'] {
+    @include checkbox;
+  }
+
+  [type='color'] {
+    @include input;
+    padding: 0;
+    min-height: 2em;
+    width: 3em;
+    @media (min-width: 600px) {
+      width: 3em;
+    }
   }
 
-  fieldset {
-    margin: 1em 0;
+  legend {
+    color: setting-get('secondary color');
+    font-weight: setting-get('bold font weight');
+  }
+
+  option {
+    margin-top: 0;
+  }
+
+  select {
+    @include input;
+    border-radius: 0;
+  }
+
+  [type='radio'] {
+    @include radio;
+  }
+
+  [type='range'] {
+    @include input;
+    height: setting-get('range height');
+    padding: 0;
+    background-color: setting-get('input border color');
+    border: 5px solid setting-get('input border color');
+    border-radius: setting-get('range height');
+    box-shadow: inset 0 0 0 0 transparent;
+
+    &:focus {
+      background-color: setting-get('primary color');
+      box-shadow: inset 0 0 0 setting-get('range outline width') setting-get('primary color');
+      outline: 0;
+    }
+
+    &:disabled {
+      opacity: .5;
+    }
+
+    &::-moz-range-track {
```

Comment:
I think it'd be worthwhile to make a `range--track` mixin like there is a `range--thumb` mixin